lib/route_downcaser/downcase_route_middleware.rb

```diff
@@ -20,6 +20,7 @@ def _call(env)
 
       # Downcase path_info if applicable
       path_info = downcased_uri(path_info)
+      path_info = path_info.gsub('%7E', '~') if path_info
```

Comment:
While I think this solution is probably a bit too simplistic, it does solve the problem. It doesn't address the other ["unreserved characters"](https://jkorpela.fi/rfc/2396/full.html#2), but I guess that's for another issue/PR to handle